packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncSessionTests.kt

```diff
@@ -570,6 +574,54 @@ class SyncSessionTests {
         }
     }
 
+    @Test
+    fun connectionState() = runBlocking {
+        Realm.open(createSyncConfig(user)).use { realm: Realm ->
+            // We don't know what the state will be, but just verify that we can retrieve it
+            // without issues
+            assertNotNull(realm.syncSession.connectionState)
+        }
+    }
+
+    @Test
+    fun connectionState_asFlow() = runBlocking {
+        Realm.open(createSyncConfig(user)).use { realm: Realm ->
+            val flow = realm.syncSession.connectionState()
+            // Adopted from realm-java tests ...
+            // Sometimes the connection is already established and then we cannot expect any
+            // updates, but as this is highly likely just safely ignore this to avoid flaky tests
+            // on CI
+            try {
+                val (oldState, newState) = withTimeout(5.seconds) { flow.first() }
+                assertNotEquals(oldState, newState)
+                assertEquals(realm.syncSession.connectionState, newState)
+            } catch (e: TimeoutCancellationException) {
+                assertTrue { realm.syncSession.connectionState == ConnectionState.CONNECTED }
+                // Make some visible sign that we have skipped waiting for events
+                println("Skipping flow tests as connection is already established")
+            }
+        }
+    }
+
+    @Test
+    fun connectionState_completeOnClose() = runBlocking {
```

Comment:
We should create an issue for adding the same behaviour to progress listeners

Reply:
It is already fixed by just wrapping the existing `progress` flow in the new [`realm.scopedFlow { ... }`](https://github.com/realm/realm-kotlin/pull/1213/files#diff-b1b23368e96f440a9e9cb5263c2b02ed80409784749dd6a354c799b5b682cac9R116). There is even [a test for it](https://github.com/realm/realm-kotlin/pull/1213/files#diff-980709ee9063ac2e05f8126cb8a69faa20f97521bbbf52808abf2e8060bfcd97R254)

Reply:
But maybe needs a note in the `progress` doc.

Reply:
Ah yeah 🙈